nifi-commons/nifi-expression-language/src/test/java/org/apache/nifi/attribute/expression/language/TestQuery.java

```diff
@@ -1177,6 +1179,99 @@ public void testReplaceAllWithMatchingGroup() {
         verifyEquals("${attr:replaceAll('.*?(l+).*', '$1')}", attributes, "ll");
     }
 
+    @Test
+    public void testReplaceShouldReplaceAllLiteralMatches() {
+        int n = 3;
+        final String ORIGINAL_VALUE = "Hello World";
+        final Map<String, String> attributes = Map.of("single", ORIGINAL_VALUE,
+                "repeating", StringUtils.repeat(ORIGINAL_VALUE, " ", n));
+        final String REPLACEMENT_VALUE = "Goodbye Planet";
+        final String EXPECTED_REPEATING_RESULT = StringUtils.repeat(REPLACEMENT_VALUE, " ", n);
+        final String REPLACE_SINGLE_EXPRESSION = "${single:replace('" + ORIGINAL_VALUE + "', '" + REPLACEMENT_VALUE + "')}";
+        final String REPLACE_REPEATING_EXPRESSION = "${repeating:replace('" + ORIGINAL_VALUE + "', '" + REPLACEMENT_VALUE + "')}";
+        Query replaceSingleQuery = Query.compile(REPLACE_SINGLE_EXPRESSION);
+        Query replaceRepeatingQuery = Query.compile(REPLACE_REPEATING_EXPRESSION);
+
+        QueryResult<?> replaceSingleResult = replaceSingleQuery.evaluate(new StandardEvaluationContext(attributes));
+        QueryResult<?> replaceRepeatingResult = replaceRepeatingQuery.evaluate(new StandardEvaluationContext(attributes));
+
+        assertEquals(REPLACEMENT_VALUE, replaceSingleResult.getValue());
+        assertEquals(AttributeExpression.ResultType.STRING, replaceSingleResult.getResultType());
+        assertEquals(EXPECTED_REPEATING_RESULT, replaceRepeatingResult.getValue());
+        assertEquals(AttributeExpression.ResultType.STRING, replaceRepeatingResult.getResultType());
+    }
+
+    @Test
+    public void testReplaceFirstShouldOnlyReplaceFirstRegexMatch() {
+        // Arrange
```

Comment:
```suggestion
```

Reply:
@exceptionfactory I made the changes you requested.